client/src/components/core/TeamSelector/useTeamSelector.tsx

```diff
@@ -0,0 +1,37 @@
+import * as React from "react";
+import { useService } from "../../../service";
+import { useStore } from "../../../store";
+import { ActionType } from "../../../store/store";
+
+export function useTeamSelector() {
+  const { fetchTeams } = useService();
+
+  const {
+    dispatch,
+    state: { teams },
+  } = useStore();
+
+  const requestInitialData = async () => {
+    // This updates the store
+    await fetchTeams();
+  };
```

Comment:
This would probably not normally be in a selector component - but this is our only component and it works - for now at least 🤷